examples/todo/test/acceptance/application.acceptance.ts

```diff
@@ -19,6 +20,21 @@ describe('Application', () => {
   before(givenAnApplication);
   before(async () => {
     await app.boot();
+
+    /**
+     * Override DataSource to not write to file for testing. Since we aren't
+     * persisting data to file and each injection normally instatiates a new
+     * instance, we must change the BindingScope to a singleton so only one
+     * instance is created and used for all injections (preserving access to
+     * the same memory space).
+     */
+    app.bind('datasources.config.db').to({
+      name: 'db',
+      connector: 'memory',
+    });
+    app.find('datasources.db')[0].inScope(BindingScope.SINGLETON);
```

Comment:
See my earlier comment - `BindingScope.SINGLETON` must be applied to all datasources by `@loopback/boot`.